src/Stripe.Tests.XUnit/BaseStripeTest.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using Xunit;
+
+namespace Stripe.Tests.Xunit
+{
+    public class BaseStripeTest : IDisposable
+    {
+        /// <value>Minimum required version of stripe-mock</value>
+        /// <remarks>
+        /// If you bump this, please don't forget to bump STRIPE_MOCK_VERSION in appveyor.yml.
+        /// </remarks>
+        private const string MockMinimumVersion = "0.19.0";
+
+        private static string Port;
+
+        private string OrigApiBase;
+        private string OrigUploadBase;
+        private string OrigApiKey;
+
+        public BaseStripeTest()
+        {
+            if (string.IsNullOrEmpty(Port))
+            {
+                CheckStripeMock();
+            }
+            SetUpStripeMock();
+        }
+
+        public void Dispose()
+        {
+            TearDownStripeMock();
+        }
+
+        protected static string GetFixture(string path)
+        {
+            return GetFixture(path, null);
+        }
+
+        protected static string GetFixture(string path, string[] expansions)
+        {
+            string url = $"http://localhost:{Port}{path}";
+
+            if (expansions != null)
+            {
+                string query = string.Join("&", expansions.Select(x => $"expand[]={x}").ToArray());
+                url += $"?{query}";
+            }
+
+            using (HttpClient client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization
+                    = new System.Net.Http.Headers.AuthenticationHeaderValue(
+                        "Bearer",
+                        "sk_test_123"
+                    );
+
+                HttpResponseMessage response;
+
+                try
+                {
+                    response = client.GetAsync(url).Result;
+                }
+                catch (Exception)
+                {
+                    throw new Exception(
+                        $"Couldn't reach stripe-mock at `localhost:{Port}`. "
+                        + "Is it running? Please see README for setup instructions."
+                    );
+                }
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    throw new Exception(
+                        $"stripe-mock returned status code: {response.StatusCode}."
+                    );
+                }
+
+                return response.Content.ReadAsStringAsync().Result;
```

Comment:
This only supports stripe-mock fixtures right? We should add a JSON one like in Java, those really are the most important ones to have/test.

Reply:
Yep. Ideally I'd like `BaseStripeTest` to have the same helper methods as we have in stripe-java, so it's still missing things like:
- retrieving contents of a local fixture file
- stubbing requests
- verifying the requested URL / parameters

The last two will be fairly complex to implement (and will likely require using a library like [moq](https://github.com/moq/moq4)) but the first one shouldn't be too hard.